state-synchronizer/src/tests/on_chain_config_tests.rs

```diff
@@ -199,4 +199,77 @@ fn test_on_chain_config_pub_sub() {
         None,
         "did not expect reconfig update"
     );
+
+    ////////////////////////////////////////////////////////////////
+    // Case 3: test failed on-chain reconfig publish throws error //
+    ////////////////////////////////////////////////////////////////
+    // test error throwing on failed reconfig publish
+    drop(reconfig_receiver);
+
+    // Create a dummy block prologue transaction that will bump the timer.
+    let txn6 = encode_block_prologue_script(gen_block_metadata(3, validator_account));
+    let txn7 = get_test_signed_transaction(
+        genesis_account,
+        /* sequence_number = */ 3,
```

Comment:
kinda surprised we don't use a variable for this instead and increment it afterward